certdehydrate/certdehydrate.go

```diff
@@ -0,0 +1,231 @@
+package certdehydrate
+
+import "bytes"
+import "crypto/sha256"
+import "crypto/x509/pkix"
+import "encoding/base64"
+import "encoding/binary"
+import "fmt"
+import "math/big"
+import "time"
+
+import "github.com/hlandau/ncdns/x509"
+
+// TODO: add a version field
+type DehydratedCertificate struct {
+	PubkeyB64 string
+	NotBeforeScaledInt int64
```

Comment:
avoid Hungarian notation: `NotBeforeScaled`, not `NotBeforeScaledInt`. Use comments on individual fields to document the format.